Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -462,18 +462,25 @@ void BlockFormattingContext::resolve_used_height_if_treated_as_auto(Box const& b
         height = max(height, calculate_inner_height(box, available_space.height, computed_values.min_height()));
     }
 
+    auto writing_mode = box.computed_values().writing_mode();
```

Comment:
let's add a layout test for this change in BFC

Reply:
This won't pass CI right now because of the test, but I added in `Tests/LibWeb/Layout/input/block-and-inline/bfc-resolve-height-if-treated-as-auto.html` trying to work from `Tests/LibWeb/Layout/input/block-and-inline/margin-must-not-collapse-across-nested-bfc.html`. The output here is just copied straight from there, wasn't sure how to fill out all the numbers. I think the test itself is mostly doing the right thing but that should definitely be checked too